From 946032d0ad19d235d78141e864f41daa73106838 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Tue, 2 Jul 2024 17:56:28 +0200
Subject: [PATCH 0987/1695] drivers: stm32_uart: rely on DT for initial console
 pins secure state

Do not call stm32_pinctrl_set_secure_cfg() at UART init as we should
rely on the device tree configuration for the pin initial
secure configuration.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1099ac13c0e0ef752f17dcd0813de839a2ec2aeb drivers: stm32_uart: rely on DT for initial console pins secure state]
Change-Id: I90ef9edc55340997a981399c8fdb349b23fd2fa5
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/391134
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_uart.c | 19 -------------------
 1 file changed, 19 deletions(-)

diff --git a/core/drivers/stm32_uart.c b/core/drivers/stm32_uart.c
index 55b847e13..c90abc1ef 100644
--- a/core/drivers/stm32_uart.c
+++ b/core/drivers/stm32_uart.c
@@ -130,31 +130,12 @@ void stm32_uart_init(struct stm32_uart_pdata *pd, vaddr_t base)
 
 static void register_secure_uart(struct stm32_uart_pdata *pd)
 {
-#if !(defined(CFG_STM32MP15) || defined(CFG_STM32MP13))
-	stm32_pinctrl_set_secure_cfg(pd->pinctrl, true);
-	if (pd->pinctrl_sleep)
-		stm32_pinctrl_set_secure_cfg(pd->pinctrl, true);
-#else
 	stm32mp_register_secure_periph_iomem(pd->base.pa);
-	stm32mp_register_secure_pinctrl(pd->pinctrl);
-	if (pd->pinctrl_sleep)
-		stm32mp_register_secure_pinctrl(pd->pinctrl_sleep);
-#endif
 }
 
 static void register_non_secure_uart(struct stm32_uart_pdata *pd)
 {
-#if !(defined(CFG_STM32MP15) || defined(CFG_STM32MP13))
-	stm32_pinctrl_set_secure_cfg(pd->pinctrl, false);
-	if (pd->pinctrl_sleep)
-		stm32_pinctrl_set_secure_cfg(pd->pinctrl, false);
-#else
 	stm32mp_register_non_secure_periph_iomem(pd->base.pa);
-	stm32mp_register_non_secure_pinctrl(pd->pinctrl);
-	if (pd->pinctrl_sleep)
-		stm32mp_register_non_secure_pinctrl(pd->pinctrl_sleep);
-#endif
-
 }
 
 static TEE_Result stm32_uart_update_baudrate(struct stm32_uart_pdata *pd,
-- 
2.47.2

